### ISSUE TITLE
Query: How book is generated?

### ISSUE THREAD

**Issue by bohrasankalp** [halfrost/LeetCode-Go#258]
Query: How book is generated? Could you please help me out. Is it autogenerated?

**Comment by halfrost** [comment]
You can use ctl tool to generate the book. Ctl tool is in the ctl file folder. You need to build this tool first and use `pdf` command to generate the whole book, which is a markdown version. Finally, you can convert the markdown version to any format you want.

**Comment by halfrost** [comment]
Yes, it is auto-generated. The site is built with [Hugo](https://gohugo.io/) using a customized `hugo-book` theme (see the `website/` directory), and the per-problem Markdown pages are generated from the Go solutions by the tooling under `ctl/`. Deployment is automated via GitHub Actions (`.github/workflows/deploy.yml`), which runs `hugo` and rsyncs the output to the server.
